manifests/charts/istio-control/istio-discovery/values.yaml

```diff
@@ -28,6 +28,11 @@ defaults:
     # Set to `type: RuntimeDefault` to use the default profile if available.
     seccompProfile: {}
 
+    # keep in sync with settings used when installing the Istio CNI chart
+    cni:
+      enabled: false
+      chained: true
```

Comment:
Fully agree this is a mess as it sits, TY for taking it up.

- I would rather rename `pilot.cni.enabled` to something that actually reflects what it does - ``pilot.inject_redirect_validation` or something.

- `pilot.cni.chained` has nothing to do with CNI chaining really, and everywhere it is used, it is only used to inject (or not) a Multus-specific network annotation AFAICT: `k8s.v1.cni.cncf.io/networks` - it should be generalized to something like `pilot.cni_networks_annotation`

Reply:
> I would rather rename pilot.cni.enabled to something that actually reflects what it does - ``pilot.inject_redirect_validation` or something.

From user point of view though, they should enable if CNI is enabled? Its kind of declarative vs imperative semantics I suppose

Reply:
They shouldn't enable one and disable the other, no - but we can't stop them from doing that today, and we don't use a parent chart or globals to ensure these values are in sync (`istioctl` may, but that's project convention at most, and if we are checking that out of band it doesn't matter if they have the same child key name anyway.)

Reply:
[I replied to your comment on the issue](https://github.com/istio/istio/issues/49290#issuecomment-1943259470), but to summarize here: I do think the most concise description of what the switch does is to enable usage of CNI, because it doesn't just trigger injection of the validation init container, but also disable the iptables init container, rendering the proxy unusable if you don't have CNI installed.

Reply:
@dgn KK, if you feel leaving the `cni` flag as-is is the most obvious I'm happy to leave that one, and either drop (or rename) `chained`.

Reply:
@bleggett I will push a PR shortly that removes `chained` form the `openshift` profile

Reply:
That said, I agree that `istio_cni.chained: true` really should probably be called `istio_cni.provider: multus`. I can make the change, but I'd prefer not to have too many things in flight, so after this PR would be best

Reply:
@dgn I'm OK with that, thank you!